sql: correct ZM coordinates for geometry_columns/geography_columns

otan opened the pull request:
Release note (bug fix): Correct how the `type` displays for ZM shapes
geometry_columns to match PostGIS output. This previously incorrectly
included the Z/M lettering.

Release note (bug fix): Correct how `type` displays in geography_columns
to better match PostGIS. This was previously in the wrong casing.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/70856)
<!-- Reviewable:end -->

otan committed:
sql: correct ZM coordinates for geometry_columns/geography_columns

Release note (bug fix): Correct how the `type` displays for ZM shapes
geometry_columns to match PostGIS output. This previously incorrectly
included the Z/M lettering.

Release note (bug fix): Correct how `type` displays in geography_columns
to better match PostGIS. This was previously in the wrong casing.

otan commented:
bors r=rafiss

this won't backport cleanly, but ill fix it later

otan commented:
blathers backport 21.1

otan commented:
blathers backport 21.1

otan commented:
blathers backport 21.2

otan commented:
blathers backport 21.2

otan commented:
blathers backport 21.2
